cosmos/models/Workflow.py

```diff
@@ -443,6 +446,19 @@ def delete(self, delete_files=False):
         self.session.commit()
         print >> sys.stderr, '%s Deleted' % self
 
+    def exit_status(self):
+        """
+        Return the exit status of the first failed Task (topologically, not chronologically).
+
+        If no Task failed, return None.
+        """
+        for t in self.task_graph():
+            if t.exit_status:
+                self.log.warning("%s took exit status %s from %s (%s)",
+                                 self, t.exit_status, t, t.status)
+                return t.exit_status
+        return None
+
```

Comment:
I think this makes for self documenting calling code:
if not workflow.successful:
  workflow.log('first failure's exit status %s ' % workflow.get_first_failed_task().exit_status)

sorry if i'm being a too nit picky!